pandas/core/indexes/multi.py

```diff
@@ -1078,6 +1078,27 @@ def levshape(self) -> Shape:
 
     @property
     def codes(self) -> tuple:
+        """
+        Codes of the MultiIndex.
+
+        Codes are the integer labels for each level designating the position of the index value in a sorted list of all index values in that level.
+
+        Returns
+        -------
+        tuple of (numpy.ndarray,)
```

Comment:
Just `tuple of numpy.ndarray`?

Reply:
On the [pandas docstring guide](https://pandas.pydata.org/docs/dev/development/contributing_docstring.html#section-3-parameters), I interpreted the section "For complex types, define the subtypes. For dict and tuple, as more than one type is present, we use the brackets to help read the type (curly brackets for dict and normal brackets for tuple)" to mean it should be tuple of (numpy.ndarray,) instead of tuple of numpy.ndarray. I am slightly confused though because it is a tuple of only one type.

One of the examples they listed was "tuple of (str,)"

Reply:
I believe `tuple of (str,)` means a tuple of length one containing a string.

Reply:
Also see the last example in that bullet list: `set of str`